supabase/functions/process-document/index.ts

```
@@ -164,33 +365,78 @@ serve(async (req) => {
-      return new TextDecoder().decode(buffer)
+      return new TextDecoder('utf-8').decode(buffer)
 
     case 'docx':
-      // Would use a Word document parser
-      return `[DOCX TEXT] - Content from ${fileName} (DOCX parsing not implemented in this demo)`
+      // For now, return placeholder - would implement proper DOCX parsing
+      return `[DOCX parsing not yet implemented for ${fileName}]`
+
+    case 'rtf':
+      // Basic RTF text extraction (simplified)
+      const rtfText = new TextDecoder('utf-8').decode(buffer)
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)